library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsMediaPeriod.java

```diff
@@ -470,20 +470,25 @@ private void buildAndPrepareSampleStreamWrappers(long positionUs) {
             : Collections.emptyMap();
 
     boolean hasVariants = !masterPlaylist.variants.isEmpty();
+    boolean hasIFrameVariants = !masterPlaylist.iFrameVariants.isEmpty();
     List<Rendition> audioRenditions = masterPlaylist.audios;
     List<Rendition> subtitleRenditions = masterPlaylist.subtitles;
 
     pendingPrepareCount = 0;
     ArrayList<HlsSampleStreamWrapper> sampleStreamWrappers = new ArrayList<>();
     ArrayList<int[]> manifestUrlIndicesPerWrapper = new ArrayList<>();
 
-    if (hasVariants) {
+    ArrayList<Variant> allVariants = new ArrayList<>();
+    allVariants.addAll(masterPlaylist.variants);
+    allVariants.addAll(masterPlaylist.iFrameVariants);
```

Comment:
I originally did it this way, https://github.com/google/ExoPlayer/issues/474#issuecomment-420594414  

@ojw28 Made good points for including all the tracks in the same adaption set https://github.com/google/ExoPlayer/issues/474#issuecomment-514307811  So basically we are adapting based on frame rate as well as bandwidth.  

One advantage to the current approach, that Oliver advocated, is it may make more sense to trick play a very low bandwidths 30FPS track rather than a higher resolution low frame rate i-Frame only track.  Having the tracks all in an adaptive set allows this to occur when adapting to speed changes without the overhead of a full on track selection.

I can go either way, could we get the parsing part in first at least and do this as a later enhancement?   Rebasing my collection of pull requests that alter this code is becoming laborious ;-), there is a lot of change to these files.